### DISCUSSION TITLE
Export to CSV

### DISCUSSION THREAD

**Discussion by CyrusV5** [devmount/third-stats#325] [Ideas]
As seen in the ThirdStats reviews, someone already suggested an export to CSV option in addition to json.
Creating this discussion here as there wasn't one.
Thank you!

**Comment by devmount** [comment]
Thanks for adding this topic. As already commented on #233, this requires a detailed description of the data structure, that's needed. How should the displayed data be exported in CSV?

**Reply by CyrusV5** [reply]
I'm sorry, didn't see that issue. Do you want me to reply here or there? Thanks.

**Reply by devmount** [reply]
No problem at all. Please reply here. Once we nailed this feature request down, I'll convert this discussion into a new issue and link it to the old one.

**Reply by alanst03** [reply]
Hi, we were thinking that the data should contain all of the existing email categories so we can filter them afterwards either by subject, correspondent or date.

**Reply by devmount** [reply]
@alanst03 So what exactly would your columns be? Please provide a corresponding table structure with some exampe data rows.

**Reply by alanst03** [reply]
@devmount Please find attached a rough example of how the export should look like.
[export example.xlsx](https://github.com/devmount/third-stats/files/6831043/export.example.xlsx)

**Reply by devmount** [reply]
@alanst03 Thank you for the example. Unfortunately this is still not sufficient for me to understand, how the data needs to be processed/grouped to fit in this table. Can you please fill in some example rows? What do you mean by "subject"? Most likely not an email subject 😅

**Reply by alanst03** [reply]
Hi and apologies for taking this long. You were right, the "subject" field is not relevant here. I've attached a new example below.
Thanks!
[export example.xlsx](https://github.com/devmount/third-stats/files/7083890/export.example.xlsx)

**Reply by devmount** [reply]
No worries. Thank you for the new example file. I think the main problem here is how the date column is supposed to be filled. What does one row represent? A single date? A date range? It's also possible to make this customizable by the user, but I'd like to keep it simple for now and see, if more users are interested in a feature like that.

**Comment by m342** [comment]
hi, sorry if my message is irrelevant but I do agree for the need (have no skill to make it here). 
I d'like to make a visual on a common format (ods, csv, whatever) in order to make visible my activity for not « computer people » . 
json is use for « computer people » but need skills to make ok for the ones who do not have theses. 
by the way thanks for the software !
In fact i d like to product rapport about my activity against stats collected by ThirdStats. 

Context : 
I have to report my activity and to be able to show this in a common format - not json :) as it need capabilities do read it)

**Reply by devmount** [reply]
Thanks for showing your interest in this. As stated before: I need some information, how the data should be structured exactly in CSV. Unlike JSON we don't have any hierarchical information in CSV. Please provide a corresponding table structure with some example data rows, so that we can discuss this as a basis.
